src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,274 @@
+package duke;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.Ui.Commands;
+import duke.Ui.Descriptors;
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DateCommand;
+import duke.command.DeleteCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+
+/**
+ * Deals with making sense of user commands and inputs.
+ */
+public class Parser {
+
+    /**
+     * Reads user input and returns the corresponding Command.
+     *
+     * @param userInput User's input into Duke chatbot.
+     * @return Command that can be executed to perform a set of actions.
+     */
+    public static Command parse(String userInput) {
+        // Only exactly "list" will be accepted for ListCommand.
+        if (userInput.equals(Commands.LIST.getCommand())) {
+            // Print tasks
+            return new ListCommand(userInput);
+        } else {
+            if (userInput.startsWith(Commands.DONE.getCommand())) {
+                // Mark task as done.
```

Comment:
Perhaps a comment might not be so necessary here? I think the way you have written your code is quite understandeable without comments!

Reply:
Thanks for pointing this out! This comment does seem to be stating the obvious and could instead contribute to clutter in the code. I will remove it in the next push.